app/models/course.rb

```diff
@@ -20,15 +21,20 @@
 #  updated_at                :datetime         not null
 #  changed_at                :datetime         not null
 #  accrediting_provider_code :text
-#  accrediting_provider_id   :integer
+#  discarded_at              :datetime
 #
 
 class Course < ApplicationRecord
+  include Discard::Model
   include WithQualifications
   include ChangedAt
 
   after_initialize :set_defaults
 
+  before_discard do
+    raise "You cannot delete a running course" if ucas_status != :new
```

Comment:
Should we pass any course info into this error? i.e. name and code

Reply:
Perhaps if we implement different ways of using it in the future, like bulk deleting. Otherwise, the system and user should be fully aware of what course it's calling `discard` on.

Reply:
I guess for logging purposes, that'd be neat.

Reply:
@dankmitchell providing provider_id and course name and code (Course.to_s). Will that be enough?